game_manager.py

```diff
@@ -0,0 +1,143 @@
+            return True
+        return False
+
+
+class Stage(object):
+    def __init__(self, index):
+        self.open = True
+        self.stage = index
+        self.champions = []
+        self.prev_champion_lists = []
+
+    def isOpen(self):
+        return self.open
+
+    def getChampions(self):
+        if not self.open and self.champions:
+            return self.champions
+
+    def setChampions(self, champions):
+        '''
+        sets the champions. need same champions for three frames to close the round
+        returns True when round closed
+        '''
+        # we need three iterations with the same champions to close the stage
+        self.prev_champion_lists.append(champions)
+        # check if lists are the same
+        if len(self.prev_champion_lists) == 3:
+            if self.prev_champion_lists[0] == self.prev_champion_lists[1] and self.prev_champion_lists[1] == self.prev_champion_lists[2]:
+                self.champions = self.prev_champion_lists[0]
+                self.open = False
+                return True
+            else:
+                self.prev_champion_lists.pop(0)
+        return False
+
+    def __str__(self):
+        output = 'Stage {} : '.format(self.stage)
+        if self.open:
+            output += 'open'
+        else:
+            output += '['
+            for champ in self.champions:
+                if champ is not None:
+                    output += str(champ) + ', '
+            output += ']'
+        return output
+
+
+class GameManager(object):
+    def __init__(self):
+        self.initialized = False
+        self.stages = []
+        self.damage_window = None
+        self.window_open = False
+
+        # set up detectors
+        self.detectors = {
+            'champion': None,
+            'stars': None
+        }
+
+    def processFrame(self, img):
+        if not self.initialized:
+            self.__initialize(img)
+        
+        if self.initialized:
+            # if window was closed and then opened create a new stage
+            window_open = self.damage_window.isOpen(img)
+            if not self.window_open and window_open:
+                self.stages.append(Stage(len(self.stages) + 1))
+            self.window_open = window_open
+
+            if self.window_open and self.stages[-1].isOpen():
+                if self.stages[-1].setChampions(self.__detectChampions(img)):
+                    print(self.stages[-1])
+
+    def draw(self, img):
+        if not self.initialized:
+            cv2.putText(img, 'NOT INITIALIZED', (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
+        elif not self.window_open:
+            cv2.putText(img, 'WINDOW CLOSED', (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
+        else:
+            self.damage_window.draw(img)
+            stage_output = str(self.stages[-1])
+            cv2.putText(img, stage_output, (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
```

Comment:
Nitpicky, but you could DRY the putText function calls with a private method

Reply:
for now gonna merge it this way but will keep in mind during next iteration